pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -593,6 +593,21 @@ def _init_strategy(self) -> None:
         else:
             raise RuntimeError(f"{self.strategy} is not valid type: {self.strategy}")
 
+        from pytorch_lightning.utilities import _IS_INTERACTIVE
+
+        is_interactive_compatible = (
+            self.strategy.is_interactive_compatible if hasattr(self.strategy, "is_interactive_compatible") else False
+        )
+        if _IS_INTERACTIVE and not is_interactive_compatible:
```

Comment:
do the _IS_INTERACTIVE check first. you can return early if it's not interactive. then you only need to check the strategy's property afterward instead of always checking it.

Reply:
We need to have strategy object to get the is_interactive_compatible property.

Reply:
I think Ananath means:

```python
if _IS_INTERACTIVE and hasattr(self.strategy, "is_interactive_compatible") and not self.strategy.is_interactive_compatible:
    raise ...
```

Reply:
```py
if not _IS_INTERACTIVE:
    return

if hasattr(self.strategy, "is_interactive_compatible") and not self.strategy.is_interactive_compatible:
    raise
```